src/robotFactory.js

```diff
@@ -1,10 +1,99 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
```

Comment:
```suggestion
  constructor(
    name,
    weight,
    {
      x = 0,
      y = 0
    },
    chipVersion
  ) {
```
pay attention to the formatting